tests/Integration/Behaviour/Features/Scenario/Product/Shop/shop_management.feature

```diff
@@ -0,0 +1,100 @@
+
+  Background:
+    Given shop "shop1" with name "test_shop" exists
+    And shop group "default_shop_group" with name "Default" exists
+    And I add a shop "shop2" with name "default_shop_group" and color "red" for the group "default_shop_group"
+    And I add a shop group "shopGroup2" with name "test_second_shop_group" and color "green"
+    And I add a shop "shop3" with name "test_third_shop" and color "blue" for the group "test_second_shop_group"
+    And I add a shop "shop4" with name "test_shop_without_url" and color "blue" for the group "test_second_shop_group"
+    And single shop context is loaded
+    Given I add product "product1" with following information:
+      | name[en-US] | magic staff |
+      | type        | standard    |
+    Then product product1 is associated to shop shop1
+    When I update product "product1" prices with following information:
+      | price              | 100.99          |
+      | ecotax             | 0               |
+      | tax rules group    | US-AL Rate (4%) |
+      | on_sale            | true            |
+      | wholesale_price    | 70              |
+      | unit_price         | 10              |
+      | unity              | bag of ten      |
+    Then product product1 should have following prices information for shops "shop1":
+      | price              | 100.99          |
+      | price_tax_included | 105.0296        |
+      | ecotax             | 0               |
+      | tax rules group    | US-AL Rate (4%) |
+      | on_sale            | true            |
+      | wholesale_price    | 70              |
+      | unit_price         | 10              |
+      | unity              | bag of ten      |
+      | unit_price_ratio   | 10.099          |
+    And product product1 is not associated to shop shop2
+    And product product1 is not associated to shop shop3
+    And product product1 is not associated to shop shop4
+
+  Scenario: I copy product to another shop that was not associated
+    # Copy values to another shop which was not associated yet
+    When I copy product product1 from shop shop1 to shop shop2
+    Then product product1 is associated to shop shop2
+    And product product1 should have following prices information for shops "shop1,shop2":
+      | price              | 100.99          |
+      | price_tax_included | 105.0296        |
+      | ecotax             | 0               |
+      | tax rules group    | US-AL Rate (4%) |
+      | on_sale            | true            |
+      | wholesale_price    | 70              |
+      | unit_price         | 10              |
+      | unity              | bag of ten      |
+      | unit_price_ratio   | 10.099          |
+    And product product1 is not associated to shop shop3
+    And product product1 is not associated to shop shop4
+    # Now modify and copy the values but this time the shop is already associated so it is an update
+    When I update product "product1" prices with following information:
```

Comment:
How can we know that the current shop is `shop1` ?

Reply:
This changed, now the shopConstraint parameter is mandatory for this command, though I kept the old behat endpoint (to keep the regular single shop scenarios simpler), but what it does behind the scene is get the default shop from the configuration